src/shared/lib/redux/types.ts

```diff
@@ -0,0 +1,12 @@
+import { Action, ThunkDispatch } from '@reduxjs/toolkit';
+import { useDispatch } from 'react-redux';
+
+import { reduxStore } from '@app/app/providers';
+
+export type RootState = ReturnType<typeof reduxStore.getState>;
+
+export type AppDispatch = typeof reduxStore.dispatch;
+
+export type ThunkAppDispatch = ThunkDispatch<RootState, void, Action>;
+
+export const useThunkDispatch = () => useDispatch<ThunkAppDispatch>;
```

Comment:
I would rename it to useAppDispatch because useThunkDispatch implies using thunks but we don't use thunks here.

Reply:
Also, it is a good practice to create useAppSelector hook.
Both of them you can move to shared/lib/hooks